src/core/frames/frame_controller.ts

```diff
@@ -427,17 +427,43 @@ export class FrameController
   }
 
   private handleFrameMissingFromResponse(fetchResponse: FetchResponse) {
-    this.renderFrameMissingError()
-    this.throwFrameMissingError(fetchResponse)
+    if (this.responsePermittedToEscapeFrame(fetchResponse)) {
+      this.warnResponseEscapingFrame(fetchResponse)
+      this.visitResponse(fetchResponse.response)
+    } else {
+      this.renderFrameMissingError()
+      this.throwFrameMissingError(fetchResponse)
+    }
+  }
+
+  private responsePermittedToEscapeFrame(fetchResponse: FetchResponse) {
+    const allowedPathsTag = this.element.ownerDocument.querySelector<HTMLMetaElement>(
+      `meta[name=turbo-frame-escape-paths]`
+    )
```

Comment:
Since this is a global setting we're reading from the current page, it seems like we ought to access it through the `session`'s `PageSnapsot`. It already has support for getting the document's meta values, which we can do at a low level like this:

```js
this.session.snapshot.getSetting("turbo-frame-escape-paths")
```

But that's private API. It would be better to provide a parsed version on `PageSnapshot`, like we have for `rootLocation` and others. Then we can depend on an `Array` return value and reduce to:

```js
return this.session.snapshot.frameEscapePaths.includes(pathname)
```

Reply:
Ah, nice. I hadn't noticed that! Fixed in https://github.com/hotwired/turbo/pull/864/commits/d64b279101bcc9ecaed4831ac4a30d3821afb641